plugin/src/frontend.ml

```diff
@@ -179,6 +181,28 @@ let lift_by_ornament ?(suffix=false) n d_orn d_orn_inv d_old =
   else
     lift_definition_by_ornament env sigma n_new l c_old
 
+(*
+  * Lift each module element (constant and inductive definitions) along the given
+  * ornament, defining a new module with all the transformed module elements.
+  *)
+let lift_module_by_ornament ident d_orn d_orn_inv mod_ref =
+  let mod_body =
+    qualid_of_reference mod_ref |> Nametab.locate_module |> Global.lookup_module
+  in
+  let lift_global gref =
+    let ident = Nametab.basename_of_global gref in
+    try
+      lift_by_ornament ident d_orn d_orn_inv (expr_of_global gref)
+    with _ ->
+      Feedback.msg_warning (str "Failed to lift " ++ pr_global_as_constr gref)
+  in
+  let _ =
+    declare_module_structure
+      ident
+      (fun _ -> iter_module_structure_by_glob lift_global mod_body)
+  in
+  Feedback.msg_notice (str "Defined lifted module " ++ Id.print ident)
```

Comment:
Use msg_info here (I eventually fix the rest to use these, but doing it for just the new code is nice)

Reply:
Done!